WooCommerce/Classes/POS/Controllers/PointOfSaleItemsController.swift

```diff
@@ -79,12 +82,33 @@ class PointOfSaleItemsController: PointOfSaleItemsControllerProtocol {
         }
     }
 
+    @MainActor
+    func loadInitialChildItems(for parent: POSItem) async {
+        guard case let .parentProduct(parentProduct) = parent else {
+            return
+        }
+
+        itemsViewStateSubject.send(itemsViewStateSubject.value.copy(itemsStack: itemsViewStateSubject.value.itemsStack.copy(itemStates: [parent: .loading([])])))
```

Comment:
This state copying/updating is quite difficult to read, and will probably contribute to more bugs like the above. Do you have any preferences for how we improve it?

Initial ideas:
1. Extract it to a function, e.g. `updateState(for: parent, to: .loading)`. It'll still be complex but at least only in one place.
2. Make it an `@Published` var, so we can simply mutate it. I didn't like this initially because I thought it might make `Observation` harder to get right when we move to it, but actually it probably maps better to observation, which uses simpler structs and vars anyway.

Reply:
Similar to above, addressed in https://github.com/woocommerce/woocommerce-ios/commit/ecf1a776db125363c49b93e56ec8de18662ecb43 https://github.com/woocommerce/woocommerce-ios/commit/15afbaed88c35108483e677cbca0ae24846bb3e9.